src/components/Topic.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import { getArticlesByQuery } from "../api";
+import Card from "./Card";
+
+const Topic = () => {
+  const [articlesByTopic, setArticlesByTopic] = useState([]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const topic = searchParams.get("topic");
+
```

Comment:
Nice use of `searchParams` to make sure the content of the page is determined by the content of the url, so it can be shared and will lead to the correct content 🙌